modules/lib/shared/utils.d.ts

```diff
@@ -0,0 +1,19 @@
+export type EmptyObject = Record<string, never>;
+
+export type AnyObject = Record<string | number | symbol, unknown>;
+
+export type Without<T, U> = { [P in Exclude<keyof T, keyof U>]?: never };
+
+export type XOR<T, U> = T | U extends object ? (Without<T, U> & U) | (Without<U, T> & T) : T | U;
```

Comment:
I don't know how I feel about using `XOR` anymore. It makes the code a little bit difficult to read. I think I would prefer code more like this style:

```TypeScript
export interface ByPath {
  id: never;
  path: string;
}

export interface ById {
  id: string;
  path: never;
}

export type ImageUrlParams = (ById | ByPath) & {
  scale: ImageScale;
  quality?: number;
  background?: string;
  format?: string;
  filter?: string;
  type?: "server" | "absolute";
  params?: Params;
};

```